#!/bin/bash
# Option for PBS
### Job name
#PBS -N mapping_post_trimming_step

### Join queuing system output and error files into a single output file
#PBS -j oe

### Request nodes, memory, walltime. N.B. THESE ARE REQUIRED
#PBS -l select=1:ncpus=12:mem=40g

### Set a list of email addresses that receive mail from PBS
#PBS -M saveria.mazzara@gmail.com

### Specify in which case an email should be send, a=abort, b=begin, e=end
#PBS -m abe

### Account string: defines the account string associated with the job
#PBS -A mazzara

# This script performs the mapping step

## Configuration 
## --------------------

# Activate the conda environment
source activate /home/mazzara/.conda/envs/rnaseq_env

# path to the run folder
INPUT_DIR="/scratch/trimm_mapping/BCL2"
OUTPUT_DIR="/scratch/trimm_mapping/BCL2/"
GENOME_DIR="/scratch/genome/STAR/"
GTF_FILE="/scratch/genome/gencode.v34.primary_assembly.annotation.gtf"


## Generate genome indexes Step
## --------------------


SAMPLE=`basename $OUTPREFIX`
cd $INPUT_DIR/$SAMPLE


STAR --genomeDir $GENOME_DIR --runThreadN 10 --readFilesIn *trim_R1.fastq.gz *trim_R2.fastq.gz --readFilesCommand zcat --outFileNamePrefix $OUTPUT_DIR/$SAMPLE --sjdbGTFfile $GTF_FILE --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonicalUnannotated --quantMode GeneCounts --outSAMtype BAM SortedByCoordinate --limitBAMsortRAM 20000000000


# Deactivate the environment
source deactivate /home/mazzara/.conda/envs/rnaseq_env
